### PR TITLE
Dln/users

### PR DESCRIPTION
**Pull request by daniellohausen**

### CODE REVIEW

File: jumpcloud/resource_user.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+			// JumopCloud offers a lot more
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+/*
+func populateUserPayload(d *schema.ResourceData) *jcapiv1.Systemuserputpost {
```
Comment (gypsydiver) [inline jumpcloud/resource_user.go]: This could be removed.

File: jumpcloud/resource_user.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+		Firstname:                   d.Get("firstname").(string),
+		Lastname:                    d.Get("lastname").(string),
+		EnableUserPortalMultifactor: d.Get("enable_mfa").(bool),
+	}
+}*/
+
+func resourceUserCreate(d *schema.ResourceData, m interface{}) error {
+	// We receive a v2config but need a v1config to continue. So, we take the only
+	// preloaded element (the x-api-key) and populate the v1config with it.
+	configv1 := jcapiv1.NewConfiguration()
```
Comment (gypsydiver) [inline jumpcloud/resource_user.go]: We could create a v2tov1 function to keep it dry.

File: jumpcloud/resource_user.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	configv1.AddDefaultHeader("x-api-key", m.(*jcapiv2.Configuration).DefaultHeader["x-api-key"])
+	client := jcapiv1.NewAPIClient(configv1)
+
+	var payload jcapiv1.Systemuserputpost
```
Comment (gypsydiver) [inline jumpcloud/resource_user.go]: Personally, I'm not a fan of this notation how about:
```go 
payload := jcapiv1.Systemuserputpost{
		Username:                    d.Get("username").(string),
		Email:                       d.Get("email").(string),
		Firstname:                   d.Get("firstname").(string),
		Lastname:                    d.Get("lastname").(string),
		EnableUserPortalMultifactor: d.Get("enable_mfa").(bool),
}
```

File: jumpcloud/resource_user.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+	}
+	return nil
+}
+
+func resourceUserUpdate(d *schema.ResourceData, m interface{}) error {
+	configv1 := jcapiv1.NewConfiguration()
+	configv1.AddDefaultHeader("x-api-key", m.(*jcapiv2.Configuration).DefaultHeader["x-api-key"])
+	client := jcapiv1.NewAPIClient(configv1)
+
+	// The code fomr the create function is almost identical, but the structure is different :
```
Comment (gypsydiver) [inline jumpcloud/resource_user.go]: typo

**Review by gypsydiver — CHANGES_REQUESTED** [review]
Added some comments, please see #4

File: jumpcloud/resource_user.go (hunk 3 above)
Comment (daniellohausen) [reply jumpcloud/resource_user.go]: Done

**Comment by gypsydiver** [comment]
:sake:

**Review by gypsydiver — APPROVED** [review]

**Pull request merged**